src/Motor.Extensions.Hosting.Kafka/KafkaMessageConsumer.cs

```diff
@@ -76,9 +81,13 @@ public async Task ExecuteAsync(CancellationToken token = default)
                 try
                 {
                     var msg = _consumer?.Consume(token);
-                    if (msg != null && !msg.IsPartitionEOF)
+                    if (msg is { IsPartitionEOF: false })
                     {
-                        SingleMessageHandling(token, msg);
+                        var handlingTask = SingleMessageHandlingAsync(msg, token);
+                        if (_options.SynchronousMessageHandling)
```

Comment:
If we're not `await`-ing the task, then we also won't get any exceptions it might throw. So, we should be sure that there are no unhandled exceptions in `SingleMessageHandlingAsync`. (See for example [this blog post](https://markheath.net/post/async-antipatterns#ignoring-tasks) for more information.)

Reply:
Good catch! Just because I'm not too familiar with this: That problem was not introduced by this PR, right?

Reply:
I'm also not 100% sure, but I think you're right. The critical code before looked like this:

```c#
var taskAwaiter = ConsumeCallbackAsync?.Invoke(cloudEvent, token).GetAwaiter();
taskAwaiter?.OnCompleted(() =>
[...]
});
```

So this was already not waited for and the handling was done asynchronously in the background as `taskAwaiter.GetResult()` wasn't called.